draft.md

```diff
@@ -0,0 +1,99 @@
+# Draft IPLD Unixfs Spec
+
+## Basic Structure
+
+A Unixfs is either a file or a directory.
+The top level IPLD object is a CBOR map with at least two fields: `type` and `data`
+and maybe a few other such as a version string or a set of flags.
+The `type` field is either `file` or `dir`.
+
+## IPLD `file`
+
+If an IPLD file is a leaf its CID type is `raw` (0x55) and has no structure.
+Otherwise its CID type is `dag-cbor` (0x71).
+The `type` field is set to `file` and the `data` field is an CBOR array.
+Each element of the array is CBOR map with the following fields:
+
+  - `data`: link
+  - `size`: cumulative size of `data`
+  - `fsize`: (file size) cumulative size of the payload of `data`
+  
+The `fsize` field is omitted if the link is `raw` as it is the same value as size.
+
+## IPLD `dir`
+
+An IPLD `dir` represents a directory.
+Its CID type is `dag-cbor` (0x71).
+The `type` field set to `dir` and the data field is an CBOR map.
+The key of the map is a filename and is a CBOR text string encoded in UTF-8.
+The value of the map is another CBOR map with the following standard fields:
+
+  - `type`
+  - `exe`: CBOR boolean: executable bit
+  - `data`: normally a CBOR link, but can be other types depending on the value of the `type` field
+  - `size`: cumulative size of `data`
+  - `fsize`: (file size) cumulative size of the payload of `data`
+  - `fname`: CBOR byte string: original filename if it differs from the key
```

Comment:
This seems unnecessary (though I admit i've missed a lot of the conversation from over in the other issue). Why would this differ from the map key?

Reply:
Yes the discussion on #2 is rather long.  It may differ because unix filenames are not required to be UTF-8.

Reply:
@whyrusleeping two things are at play / in conflict:
- Desire for full POSIX compatibility mandates names to allow **any** sequence of bytes excluding `0x00` and `0x2f`
- The current proto-IPLD spec mandates keys ( i.e. names ) to be unicode: the mandate flows from [the spec declaring](https://github.com/ipld/specs/blob/bbb0bba8/ipld/README.md#canonical-format) a strict superset of [RFC 7049](https://tools.ietf.org/html/rfc7049#section-3.7)

Reply:
Hrm, I don't have a lot of strong opinions here. I will defer to @lgierth @diasdavid and @Stebalien

Reply:
Is this field going include which character set should be used to interpret it somehow?

Reply:
No.  Just the raw byte string if it differs from the key (which is the filename) that is in utf-8.  For display the key should be used.